firebase-sessions/src/main/kotlin/com/google/firebase/sessions/settings/SettingsCache.kt

```diff
@@ -0,0 +1,162 @@
+ */
+
+package com.google.firebase.sessions.settings
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.doublePreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.intPreferencesKey
+import androidx.datastore.preferences.core.longPreferencesKey
+import kotlinx.coroutines.flow.first
+
+internal data class SessionConfigs(
+  val sessionEnabled: Boolean?,
+  val sessionSamplingRate: Double?,
+  val cacheDuration: Int?,
+  val sessionRestartTimeout: Int?,
+  val cacheUpdatedTime: Long?
+)
+
+internal class SettingsCache(private val store: DataStore<Preferences>) {
+  private var sessionConfigs =
+    SessionConfigs(
+      sessionEnabled = null,
+      sessionSamplingRate = null,
+      sessionRestartTimeout = null,
+      cacheDuration = null,
+      cacheUpdatedTime = null
+    )
+
+  private object SettingsCacheKeys {
+    val SETTINGS_CACHE_SESSIONS_ENABLED = booleanPreferencesKey("firebase_sessions_enabled")
+    val SETTINGS_CACHE_SAMPLING_RATE = doublePreferencesKey("firebase_sessions_sampling_rate")
+    val SETTINGS_CACHE_SESSIONS_RESTART_TIMEOUT_SECONDS =
+      intPreferencesKey("firebase_sessions_restart_timeout")
+    val SETTINGS_CACHE_SESSIONS_CACHE_DURATION_SECONDS =
+      intPreferencesKey("firebase_sessions_cache_duration")
+    val SETTINGS_CACHE_SESSIONS_CACHE_UPDATED_TIME =
+      longPreferencesKey("firebase_sessions_cache_updated_time")
+  }
+
+  private suspend fun updateSessionConfigs() = mapSessionConfigs(store.data.first().toPreferences())
+
+  private fun mapSessionConfigs(settings: Preferences): SessionConfigs {
+    val sessionEnabled = settings[SettingsCacheKeys.SETTINGS_CACHE_SESSIONS_ENABLED]
+    val sessionSamplingRate = settings[SettingsCacheKeys.SETTINGS_CACHE_SAMPLING_RATE]
+    val sessionRestartTimeout =
+      settings[SettingsCacheKeys.SETTINGS_CACHE_SESSIONS_RESTART_TIMEOUT_SECONDS]
+    val cacheDuration = settings[SettingsCacheKeys.SETTINGS_CACHE_SESSIONS_CACHE_DURATION_SECONDS]
+    val cacheUpdatedTime = settings[SettingsCacheKeys.SETTINGS_CACHE_SESSIONS_CACHE_UPDATED_TIME]
+
+    sessionConfigs =
+      SessionConfigs(
+        sessionEnabled = sessionEnabled,
+        sessionSamplingRate = sessionSamplingRate,
+        sessionRestartTimeout = sessionRestartTimeout,
+        cacheDuration = cacheDuration,
+        cacheUpdatedTime = cacheUpdatedTime
+      )
+    return sessionConfigs
+  }
+
+  internal fun hasCacheExpired(): Boolean {
+    if (sessionConfigs.cacheUpdatedTime != null) {
+      val currentTimestamp = System.currentTimeMillis()
+      val timeDifferenceSeconds = (currentTimestamp - sessionConfigs.cacheUpdatedTime!!) / 1000
+      if (timeDifferenceSeconds < sessionConfigs.cacheDuration!!) return false
+    }
+    return true
+  }
+
+  fun sessionsEnabled(): Boolean? {
```

Comment:
Each of these one line functions can be replaced with the notation `fun sessionsEnabled(): Boolean? = sessionConfigs.sessionEnabled` it will make this class much shorter.

Reply:
Good suggestion. Done.